# HG changeset patch
# User Michael Goffioul <michael.goffioul@gmail.com>
# Date 1385057647 18000
#      Thu Nov 21 13:14:07 2013 -0500
# Node ID afeb233ea4430d0a9dc863dce88fa8361a702d60
# Parent  54784619350bf7c3dcdcacdbf095667c25666f8c
[Win32] Keep console shown when HIDDEN_CONSOLE is not defined.

* libgui/src/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
(QConsolePrivate::QConsolePrivate): Don't hide windows console when
HIDDEN_CONSOLE is not defined.

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -325,17 +325,19 @@ QConsolePrivate::QConsolePrivate (QWinTe
   HANDLE hStdOut = GetStdHandle (STD_OUTPUT_HANDLE);
 
   log ("Console allocated: hStdOut: %p\n", hStdOut);
 
   m_stdOut = hStdOut;
   m_consoleWindow = GetConsoleWindow ();
 
   // In case the console window hasn't been created hidden...
+#ifdef HIDDEN_CONSOLE
   ShowWindow (m_consoleWindow, SW_HIDE);
+#endif
 
   CONSOLE_SCREEN_BUFFER_INFO sbi;
 
   GetConsoleScreenBufferInfo (hStdOut, &sbi);
   m_bufferSize = QSize (sbi.dwSize.X, qMax (sbi.dwSize.Y, (SHORT)500));
   m_consoleRect = QRect (sbi.srWindow.Left, sbi.srWindow.Top,
                          sbi.srWindow.Right - sbi.srWindow.Left + 1,
                          sbi.srWindow.Bottom - sbi.srWindow.Top + 1);
